pkg/core/managers/apis/mesh/mesh_manager.go

```diff
@@ -146,6 +161,14 @@ func (m *meshManager) Update(ctx context.Context, resource core_model.Resource,
 	if err := resource.Validate(); err != nil {
 		return err
 	}
+
+	currentMesh := &core_mesh.MeshResource{}
+	if err := m.Get(ctx, currentMesh, core_store.GetBy(core_model.MetaToResourceKey(mesh.GetMeta()))); err != nil {
+		return err
+	}
+	if err := m.meshValidator.ValidateUpdate(ctx, currentMesh, mesh); err != nil {
+		return err
+	}
 	return m.store.Update(ctx, mesh, fs...)
```

Comment:
Doesn't take into account optimistic lock (`Mesh` might change in between `m.meshValidator.ValidateUpdate` and `m.store.Update`)

Reply:
Will be resolved in https://github.com/Kong/kuma/issues/487